### ISSUE TITLE
feat: As a user, I want to start Apache APISIX with a etcd user (who only have read permission of etcd), so that we can improve the security and avoid changing the etcd in DP side

### ISSUE THREAD

**Issue by starsz** [apache/apisix#6735]
### Description

Hello,  I config the Apache APISIX with a user (who only has read permission of `/apisix/`), and when I start Apache APISIX.
I meet this problem: 
```
got malformed key-put message: "{"error":"etcdserver: permission denied","code":7,"message":"etcdserver: permission denied"}" from etcd "http://127.0.0.1:2379/v3/kv/put
```

The code is as follow:
https://github.com/apache/apisix/blob/55eb58ab85ac7e108de36271fd9cf29fe2fa3324/apisix/cli/etcd.lua#L310-L365

I think we can avoid adding the `init_dir` if the directory exists.

**Comment by tzssangglass** [comment]
Here are some examples of DP writing data in etcd:

1. `request-id `plugin:https://github.com/apache/apisix/blob/542569091d4b6a40b9d560c4fe76077ee9dba087/apisix/plugins/request-id.lua#L93
2. `server-info` plugin: https://github.com/apache/apisix/blob/542569091d4b6a40b9d560c4fe76077ee9dba087/apisix/plugins/server-info.lua#L128

welcome to add.

**Comment by shuaijinchao** [comment]
I think DP should only have read-only operations, but this should be at the framework level. Plugins are generally custom and can be left to the user for decision-making.

**Comment by starsz** [comment]
> Here are some examples of DP writing data in etcd:
> 
> 1. `request-id `plugin:https://github.com/apache/apisix/blob/542569091d4b6a40b9d560c4fe76077ee9dba087/apisix/plugins/request-id.lua#L93
> 2. `server-info` plugin: https://github.com/apache/apisix/blob/542569091d4b6a40b9d560c4fe76077ee9dba087/apisix/plugins/server-info.lua#L128
> 
> welcome to add.

So if I want to config the user with read-only operation, I shouldn't use these plugins.

And I think we should also skip adding the init_dir if the directory exists.

**Comment by tzssangglass** [comment]
> So if I want to config the user with read-only operation, I shouldn't use these plugins.
> 
> And I think we should also skip adding the init_dir if the directory exists.

yes, we can add configuration to control skipping the `init_etcd` step.
